pandas/tests/io/test_sql.py

```diff
@@ -2053,6 +2072,41 @@ class Temporary(Base):
 
         tm.assert_frame_equal(df, expected)
 
+    # -- SQL Engine tests (in the base class for now)
+    def test_invalid_engine(self):
+        msg = "engine must be one of 'auto', 'sqlalchemy'"
+        with pytest.raises(ValueError, match=msg):
+            self._to_sql_with_sql_engine("bad_engine")
+
+    def test_options_sqlalchemy(self):
+        # use the set option
+
+        with pd.option_context("io.sql.engine", "sqlalchemy"):
+            self._to_sql_with_sql_engine()
+
+    def test_options_auto(self):
+        # use the set option
+
+        with pd.option_context("io.sql.engine", "auto"):
+            self._to_sql_with_sql_engine()
+
+    def test_options_get_engine(self):
+        assert isinstance(get_engine("sqlalchemy"), SQLAlchemyEngine)
+
+        with pd.option_context("io.sql.engine", "sqlalchemy"):
+            assert isinstance(get_engine("auto"), SQLAlchemyEngine)
+            assert isinstance(get_engine("sqlalchemy"), SQLAlchemyEngine)
+
+        with pd.option_context("io.sql.engine", "auto"):
+            assert isinstance(get_engine("auto"), SQLAlchemyEngine)
+            assert isinstance(get_engine("sqlalchemy"), SQLAlchemyEngine)
+
+    def test_get_engine_auto_error_message(self):
+        # Expect different error messages from get_engine(engine="auto")
+        # if engines aren't installed vs. are installed but bad version
+        pass
+        # TODO fill this in when we add more engines
```

Comment:
can you add a test here which raises if the io.sql.engine is specified but not supported

Reply:
I think that's already done in the first test here
https://github.com/pandas-dev/pandas/blob/dbf0cfa9337789a24afbd1034bee1fb69a058b57/pandas/tests/io/test_sql.py#L2076-L2079
which calls a method that passes the engine param (in this case, an invalid one named `bad_engine`) to the function
https://github.com/pandas-dev/pandas/blob/dbf0cfa9337789a24afbd1034bee1fb69a058b57/pandas/tests/io/test_sql.py#L580-L595

Does this suffice? Or did you have something else in mind?

The test that I left as `# TODO` is for ensuring that when importing 3rd party libraries for the upcoming engines it raises the correct Exceptions if it can't import them

Reply:
@jreback you marked this as resolved, assuming it's ok then?